### PR TITLE
[receiver/hostmetrics/memory] Add system.memory.usage{state=total}

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Add `system.memory.usage` with `state=total` metric to report the total RAM memory available on the system.

After open-telemetry/semantic-conventions/pull/89 this is a valid value for `system.memory.usage`.

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
Tests are failing https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6312718857/job/17139373415?pr=27096

File: receiver/hostmetricsreceiver/internal/scraper/memoryscraper/metadata.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ attributes:
   state:
     description: Breakdown of memory usage by type.
     type: string
-    enum: [buffered, cached, inactive, free, slab_reclaimable, slab_unreclaimable, used]
+    enum: [total, buffered, cached, inactive, free, slab_reclaimable, slab_unreclaimable, used]
```
Comment (crobert-1) [inline receiver/hostmetricsreceiver/internal/scraper/memoryscraper/metadata.yaml:11]:
```suggestion
    enum: [buffered, cached, free, inactive, slab_reclaimable, slab_unreclaimable, used, total]
```
This was _almost_ in alphabetical order, we may as well sort all of it.

**Comment by crobert-1** [comment]
> Tests are failing https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6312718857/job/17139373415?pr=27096

Looks like the tests are relying on the index of datapoints within the gauge, and those are now off because of the new state.

**Comment by mx-psi** [comment]
After discussion with the semantic conventions WG we are going to change the spec for this first to
- prohibit `total` as a `state` value and 
- ensure the sum of all dimensions corresponds to total

**Pull request closed without merging**